### ISSUE TITLE
Tacotron: Advanced attention module (e.g. Monotonic attention)

### ISSUE THREAD

**Issue by reuben** [mozilla/TTS#13]

**Comment by erogol** [comment]
Trying GMM Attention...

**Comment by erogol** [comment]
Using a additional loss over attention matrix to enforce it to take diagonal shape has very promising results. Model seems to converge much faster and even after 20K iter, it starts to give very good fidelity on validation set.

The loss is a L1 loss over a mask generated with exp((n/N - t/T)^2/k) where k is a constant defining the width of the diagonal mask and n, N number of characters, t,T is number of frames. 

An example mask is as below with k=0.05
![index](https://user-images.githubusercontent.com/1402048/39247115-cacc66f0-4898-11e8-9fac-edf7d302f664.png)

**Comment by erogol** [comment]
At the test time guided attention model suffers at long sequences and more frequently misaligned.

**Comment by xinqipony** [comment]
@erogol do you have a change related to guided attention?

I am thinking use phone duration information to generate the guided attention for training;

**Comment by erogol** [comment]
@xinqipony there is guided-attention branch.

**Comment by njellinas** [comment]
@erogol Hi, can you provide the code for the GMM attention? I cannot find a working version that gives good alignments anywhere.

**Comment by erogol** [comment]
@njellinas  I don't have it either anymore. I totally ditched it. You can pick that out from "voice loop" repo.
